cli/pcluster/cli_commands/compute_fleet_status_manager.py

```diff
@@ -0,0 +1,141 @@
+
+        pass
+
+    def __init__(self, cluster_name):
+        self._table_name = "parallelcluster-" + cluster_name
+        self._ddb_resource = boto3.resource("dynamodb")
+        self._table = self._ddb_resource.Table(self._table_name)
+
+    def get_status(self, fallback=None):
+        """Get compute fleet status."""
+        try:
+            compute_fleet_status = self._table.get_item(ConsistentRead=True, Key={"Id": self.COMPUTE_FLEET_STATUS_KEY})
+            if not compute_fleet_status or "Item" not in compute_fleet_status:
+                raise Exception("COMPUTE_FLEET status not found in db table")
+            return ComputeFleetStatus(compute_fleet_status["Item"][self.COMPUTE_FLEET_STATUS_ATTRIBUTE])
+        except Exception as e:
+            LOGGER.error("Failed when retrieving fleet status from DynamoDB with error %s", e)
+            return fallback
+
+    def update_status(self, current_status, next_status):
+        """Set compute fleet status."""
+        try:
+            self._table.put_item(
+                Item={"Id": self.COMPUTE_FLEET_STATUS_KEY, self.COMPUTE_FLEET_STATUS_ATTRIBUTE: str(next_status)},
+                ConditionExpression=Attr(self.COMPUTE_FLEET_STATUS_ATTRIBUTE).eq(str(current_status)),
+            )
+        except self._ddb_resource.meta.client.exceptions.ConditionalCheckFailedException as e:
+            raise ComputeFleetStatusManager.ConditionalStatusUpdateFailed(e)
+
+    def update_status_and_wait_transition(self, request_status, in_progress_status, final_status):
+        """
+        Update the status of the compute fleet and wait for a status transition.
+
+        It updates the status of the fleet to request_status and then waits for it to be updated to final_status,
+        by eventually transitioning through in_progress_status
+        """
+        compute_fleet_status = self.get_status()
+        if not compute_fleet_status:
+            raise Exception("Could not retrieve compute fleet status.")
+
+        if compute_fleet_status == final_status:
+            LOGGER.info("Compute fleet already in %s status.", final_status)
+            return
+
+        if compute_fleet_status not in {request_status, in_progress_status, final_status}:
+            self.update_status(current_status=compute_fleet_status, next_status=request_status)
+        LOGGER.info("Submitted compute fleet status transition request. Waiting for status update to start...")
+
+        compute_fleet_status = self._wait_for_status_transition(wait_on_status=request_status, timeout=180)
+        if compute_fleet_status == in_progress_status:
+            LOGGER.info(
+                "Compute fleet status transition is in progress. This operation might take a while to complete..."
+            )
+            compute_fleet_status = self._wait_for_status_transition(wait_on_status=in_progress_status, timeout=600)
```

Comment:
Timeout 600 means that the command will hang waiting for the status change for 10 minutes?
We need to add a message for the user.

Reply:
yes at most 10 mins. let me know if I need to reword the message above or if you just missed it.

Reply:
I mean we should communicate the user we're performing some actions, e.g. a message each minute, or improve the previous message saying that it could take up to 10 minutes. 
Without a message it seems that the command is hanging.